solarathon/pages/dashboard.py

```diff
@@ -87,31 +133,42 @@ def DashboardCard(price: str, procentChange: str, icon: solara.Element = None, m
                     },
                 ):
                         with solara.GridFixed(columns=2, justify_items="space-between", align_items="baseline"):
-                            solara.Text(str(f"{decimals(price)}$"), style={"font-size": "1.5rem", "font-weight": 500})
+                            solara.Text(str(f"{decimals(ticker_data.last_price)}$"), style={"font-size": "1.5rem", "font-weight": 500})
                             solara.Text(str("price"), style={"font-size": "0.6rem"})
                             if market_cap is not None: solara.Text(str(f"{format_price(decimals(market_cap))}$"), style={"font-weight": 400})
                             if market_cap is not None:  solara.Text(str("market cap"), style={"font-size": "0.6rem"})
-                            solara.Text(str(procentChange + "%"), style={"color": processColor(procentChange), "font-weight": 500})
+                            solara.Text(f"{ticker_data.price_change_percent}%", style={"color": processColor(ticker_data.price_change_percent), "font-weight": 500})
                             solara.Text(str("24h change price"), style={"font-size": "0.6rem"})
-                            if market_cap_change_percentage is not None: solara.Text(str(market_cap_change_percentage) + "%", style={"color": processColor(procentChange), "font-weight": 500})
+                            if market_cap_change_percentage is not None: solara.Text(str(market_cap_change_percentage) + "%", style={"color": processColor(ticker_data.price_change_percent), "font-weight": 500})
                             if market_cap_change_percentage is not None: solara.Text(str("24h change market cap"), style={"font-size": "0.6rem"})
 
-@solara.component
-def Page():
-    loading, set_loading = solara.use_state(True)
-    default_currency = "USDT"
-    default_echange = "Binance"
-    symbols_url = "https://api.binance.com/api/v3/exchangeInfo"
-    coingecko_json_url = "https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc"
-    binance_url = "https://api.binance.com/api/v3/ticker/24hr?symbol="
 
+def get_available_symbols():
+    symbols_url = "https://api.binance.com/api/v3/exchangeInfo"
     try:
-        symbols_response = requests.get(symbols_url, verify=False)
+        symbols_response = requests.get(symbols_url)
         symbols_data = symbols_response.json()
         available_symbols = sorted(list(set([symbol['baseAsset'].lower() for symbol in symbols_data['symbols']])))
     except Exception as e:
         available_symbols = []
 
+    return available_symbols
+
+
+@solara.component
+def Page():
+    default_currency = "USDT"
+    default_echange = "Binance"
+
+    # Store in state and make sure it won't be refetched when
+    # component is rerendered.
+    available_symbols, _ = solara.use_state(
```

Comment:
Still wondering if it makes sense to fetch this here instead of outside of the component. This list won't change very often, and is the same for each viewer of the dashboard.

Reply:
seems good with use_memo, do you mean fetching outside the component and saving the list in `solara.reactive`?